src/caret_analyze/architecture/architecture.py

```diff
@@ -207,23 +226,40 @@ def _verify(nodes: Collection[NodeStruct]) -> None:
                      f'callback_type: {uniqueness_violated[0]}'
                      f'period_ns: {uniqueness_violated[1]}'))
 
+    def rename_callback(self, src: str, dst: str):
+        cb_s: List[CallbackStruct] =\
+            Util.flatten(cb_g.callbacks for cb_g in
+                         Util.flatten([e.callback_groups for e in self._executors]))
+        c: CallbackStruct = Util.find_similar_one(src, cb_s, lambda x: x.callback_name)
+        c.callback_name = dst
+
+    def rename_node(self, src: str, dst: str):
+        for n in self._nodes:
+            n.rename_node(src, dst)
+
+        for e in self._executors:
+            e.rename_node(src, dst)
+
+        for c in self._communications:
+            c.rename_node(src, dst)
 
-"""
-    def rename_callback(src: str, dest: str):
-        raise NotImplementedError('')
+    def rename_path(self, src: str, dst: str):
+        p: PathStruct = Util.find_similar_one(src, self._paths, lambda x: x.path_name)
+        p.path_name = dst
 
-    def rename_node(src: str, dest: str):
-        raise NotImplementedError('')
+    def rename_executor(self, src: str, dst: str):
+        e: ExecutorStruct = Util.find_similar_one(src, self._executors, lambda x: x.executor_name)
+        e.executor_name = dst
 
-    def rename_path(src: str, dest: str):
-        raise NotImplementedError('')
+    def rename_topic(self, src: str, dst: str):
+        for n in self._nodes:
+            n.rename_topic(src, dst)
 
-    def rename_executor(src: str, dest: str):
-        raise NotImplementedError('')
+        for e in self._executors:
+            e.rename_topic(src, dst)
 
-    def rename_topic(src: str, dest: str):
-        raise NotImplementedError('')
-"""
+        for c in self._communications:
+            c.rename_topic(src, dst)
 
 
 class NamedPathManager():
```

Comment:
NamedPathManager is no longer needed and should be removed.

Reply:
Fixed it in [312dc18](https://github.com/tier4/CARET_analyze/pull/156/commits/312dc18d63e647938a6325e9d87b1569f2776b34)

Unit tests have been confirmed.